Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift

```diff
@@ -21,11 +26,31 @@ class LoginViewController: LoginSignupValidationMasterView {
         self.navigationItem.hidesBackButton = true
         self.navigationController?.setNavigationBarHidden(true, animated: false)
       
-        self.gradientView.colors = [UIColor.yiGrapeTwoColor(), UIColor.yiGrapeTwoColor()]
-        
         UIApplication.sharedApplication().setStatusBarStyle(UIStatusBarStyle.LightContent, animated: false)
+        
         //Get user call
         checkUserExists()
+        
+        if isFromSettings {
+            screenNameLabel.text = NSLocalizedString("change-pin", comment: "")
+            backButton.hidden = false
+            topView.backgroundColor = UIColor.yiMangoColor()
+            self.gradientView.colors = [UIColor.yiMangoTriangleColor(), UIColor.yiMangoTriangleColor()]
+            gradientContainerView.backgroundColor = UIColor.yiMangoColor()
+            
+            let viewWidth = self.view.frame.size.width
+            let customView=UIView(frame: CGRectMake(0, 0, (viewWidth-60)/3, 2))
+            customView.backgroundColor=UIColor.yiDarkishPinkColor()
```

Comment:
This is the third place I find this code snippet repeated.
Put its logic in one place (extension or global function), then re-use it instead of copying the code in many places.